addons/sourcemod/scripting/VIP_FreeVIPGiveaway.sp

```diff
@@ -107,64 +175,154 @@ public void Event_RoundStart(Event event, const char[] name, bool dontBroadcast)
+		g_hCookie.Set(client, "true");
 	}
+
+	return Plugin_Continue;
 }
 
 public void OnClientDisconnect(int client)
 {
-	if(!client)
+	if(IsClientSourceTV(client) || IsFakeClient(client))
 		return;
 
 	char testVipGroup[16];
-	GetConVarString(g_Cvar_TestVIPGroup, testVipGroup, sizeof(testVipGroup));
+	g_Cvar_TestVIPGroup.GetString(testVipGroup, sizeof(testVipGroup));
+
+	if(!VIP_IsClientVIP(client) || VIP_GetClientID(client) != -1 || VIP_GetClientVIPGroup(client, testVipGroup, sizeof(testVipGroup)))
+		return;
 
-	if(IsClientInGame(client) && VIP_IsClientVIP(client) && VIP_GetClientID(client) == -1 && !VIP_GetClientVIPGroup(client, testVipGroup, sizeof(testVipGroup)))
-		VIP_RemoveClientVIP2(_, client, false, false);
+	VIP_RemoveClientVIP2(_, client, false, false);
 }
 
-public Action Command_FreeVIP(int client, int argc)
+Action Command_FreeVIP(int client, int args)
 {
-	int playersOnServer = GetClientCount() -1; // -1 cuz of sourcetv
-	int minPlayers = GetConVarInt(g_Cvar_MinPlayers);
+	int playersOnServer = GetClientCount();
+	for(int i = 1; i <= MaxClients; i++)
+	{
+		if(i <= 0 || i > MaxClients || !IsClientConnected(i))
+			continue;
+
+		if (IsClientSourceTV(i))
+			playersOnServer = playersOnServer  - 1; // -1 cuz of sourcetv
+	}
+	int minPlayers = g_Cvar_MinPlayers.IntValue;
+	int freeVIPStart = g_Cvar_FreeVIPStart.IntValue;
+	int freeVipEnd = g_Cvar_FreeVIPEnd.IntValue;
+
+	if(freeVIPStart > GetTime() || freeVipEnd < GetTime())
+	{
+		CReplyToCommand(client, "{green}[SM] {default}Free {pink}VIP {default}Giveaway is {red}disabled{default}.");
+		return Plugin_Handled;
+	}
 
 	if (playersOnServer >= minPlayers)
 	{
-		CPrintToChat(client, "{default}Free {pink}VIP {default}Giveaway is {green}enabled{default}.");
+		CReplyToCommand(client, "{green}[SM] {default}Free {pink}VIP {default}Giveaway is {green}enabled{default}.");
+		return Plugin_Handled;
 	}
 	else
 	{
 		int playersNeeded = minPlayers - playersOnServer;
 
-		CPrintToChat(client, "{default}Free {pink}VIP {default}Giveaway is {red}disabled{default}.\nPlayers on: {green}%d {default}| Players required: {green}%d {default}| Players needed: {green}+%d", playersOnServer, minPlayers, playersNeeded);
+		CReplyToCommand(client, "{green}[SM] {default}Free {pink}VIP {default}Giveaway is {red}disabled{default}.\nPlayers on: {green}%d {default}| Players required: {green}%d {default}| Players needed: {green}+%d", playersOnServer, minPlayers, playersNeeded);
+		return Plugin_Handled;
+	}
+}
+
+void SetHostName()
+{
+	int freeVIPStart = g_Cvar_FreeVIPStart.IntValue;
+	int freeVipEnd = g_Cvar_FreeVIPEnd.IntValue;
+
+	if(freeVIPStart > GetTime())
+		return;
+
+	if(freeVipEnd < GetTime())
+		return;
+
+	if (!g_sHostname[0])
+	{
+		g_Cvar_HostNamePrefix.GetString(g_sHostnamePrefix, sizeof(g_sHostnamePrefix));
+		g_Cvar_Hostname.GetString(g_sHostname, sizeof(g_sHostname));
+
+		if (g_sHostname[0] && g_sHostnamePrefix[0] && StrContains(g_sHostname, g_sHostnamePrefix, true) == -1)

```

Comment:
put this outside of the if condition so that it checks if the hostname has changed between each map